src/site/content/en/blog/codelab-building-a-sidenav-component/index.md

```diff
@@ -0,0 +1,332 @@
+---
+layout: codelab
+title: "Codelab: Building a Sidenav component"
+authors:
+  - adamargyle
+description: |
+  Learn how to build a responsive slide out side navigation layout component.
+date: 2020-11-25
+hero: hero.jpg
+thumbnail: thumb.jpg
+glitch: gui-challenges-sidenav-lab
+glitch_path: app/index.html
+related_post: building-a-sidenav-component
+tags:
+  - css
+  - dom
+  - javascript
+  - layout
+  - mobile
+  - ux
+---
+
+This codelab teaches you how to build a responsive slide out side navigation layout component
+on the web. We'll build the component as we go, starting with HTML, then CSS,
+then JavaScript.
+
+Check out my blog post [Building a Sidenav component](/building-a-sidenav-component)
+to learn about the CSS tools chosen for building this component.
+
+{% YouTube 'PzvdREGR0Xw' %}
+
+## Setup
+
+{% Instruction 'remix', 'ol' %}
+1. Open `app/index.html`.
+
+## HTML
+
+First, get the essentials of the HTML setup so there's content and some boxes to work with.
+I'll be using an `<aside>` to hold the navigation menu, and a `<main>` to hold the primary page content. 
+Drop the following HTML into the body tag.
+
+```html
+<aside></aside>
+<main></main>
+```
+
+Next I'll fill in those semantic elements with the rest of the page content. Add a 
+navigation element, some nav links and a close link. 
+
+```html/1-14
+<aside>
+  <nav>
+    <h4>My</h4>
+    <a href="#">Dashboard</a>
+    <a href="#">Profile</a>
+    <a href="#">Preferences</a>
+    <a href="#">Archive</a>
+
+    <h4>Settings</h4>
+    <a href="#">Accessibility</a>
+    <a href="#">Theme</a>
+    <a href="#">Admin</a>
+  </nav>
+  
+  <a href="#"></a>
+</aside>
+```
+
+In the main content element, I added a header and an article to semantically hold the layout content. 
+The header will have the menu open button, we'll show and hide it based on viewport size soon. The aside 
+has the close button.
```

Comment:
We're missing a clear instruction here. You mention what you did but it's not immediately clear what the reader has to do. Of course I know that you mean that they should add this header and article element to the main element. But we shouldn't make our readers have to do a mental diff in their head to figure out what they have to do.